internal/runconfig/runconfig_test.go

```diff
@@ -1006,6 +1006,72 @@ func TestGenRunConfig(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "test runconfig generation encodedauth",
+			in: &config.Config{
+				Runs: []*config.Run{
+					&config.Run{
+						Name: "run01",
+						DockerRegistriesAuth: map[string]*config.DockerRegistryAuth{
+							"index.docker.io": {
+								Type: config.DockerRegistryAuthTypeEncodedAuth,
+								Auth: "dXNlcm5hbWU6cGFzc3dvcmQy",
+							},
+						},
+						Tasks: []*config.Task{
+							&config.Task{
+								Name: "task01",
+								DockerRegistriesAuth: map[string]*config.DockerRegistryAuth{
+									"index.docker.io": {
+										Type: config.DockerRegistryAuthTypeEncodedAuth,
+										Auth: "dXNlcm5hbWU6cGFzc3dvcmQy",
```

Comment:
Probably making this Auth different from the run docker registry auth will let us know if the overidding works.

So, for this reason I'll create 3 different tests:
* Only with the global  docker registry auth
* With both global and run docker registry auth to see if the overrides works
* With global, run and task docker registry auth to see if the overrides works